enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useState } from "react";
+
+import {
+  loadSdkQuestion,
+  type LoadSdkQuestionParams,
+} from "embedding-sdk/lib/load-question";
+import {
+  runQuestionOnNavigateSdk,
+  runQuestionOnQueryChangeSdk,
+} from "embedding-sdk/lib/run-question-query";
+import type {
+  NavigateToNewCardParams,
+  SdkQuestionResult,
+} from "embedding-sdk/types/question";
+import { useDispatch } from "metabase/lib/redux";
+import type Question from "metabase-lib/v1/Question";
+
+export function useLoadQuestion(options: LoadSdkQuestionParams) {
+  const { location, params } = options;
+
+  const dispatch = useDispatch();
+
+  const [result, setQuestionResult] = useState<SdkQuestionResult>({});
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
+
+  const { question, queryResults } = result;
+
+  const storeQuestionResult = async (
+    getQuestionResult: () => Promise<SdkQuestionResult | null>,
+  ) => {
+    setIsQuestionLoading(true);
+
+    try {
+      const result = await getQuestionResult();
+
+      if (result) {
+        setQuestionResult(result);
+      }
+    } catch (e) {
+      console.error(`Failed to update question result`, e);
```

Comment:
We seem to already handle error states for query errors as part of query results.

We could return a `<SdkError />` in case something unexpected happens (e.g. network errors) - at the moment it ignores unexpected errors.

Reply:
Maybe use `react-use`? Otherwise you need to handle hook unmounting, responses coming out of order etc

Reply:
@ranquild Thank you! To clarify, are you specifically referring to using `useAsyncFn` to ensure that only one responses can be processed at a time, so they don't go out-of-order?

I'm not sure what you mean by the "hook unmounting" part though, do you mean request cancellation when the component unmounts?

Reply:
yep I meant that that your promise could resolve when the component unmounts